lib_eio/net.ml

```diff
@@ -267,3 +269,52 @@ let with_tcp_connect ?(timeout=Time.Timeout.none) ~host ~service t f =
   | exception (Exn.Io _ as ex) ->
     let bt = Printexc.get_raw_backtrace () in
     Exn.reraise_with_context ex bt "connecting to %S:%s" host service
+
+let accept_fork_domain ~on_error (domain_mgr, domains) (t : #listening_socket) connection_handler () =
+  Switch.run @@ fun sw ->
+  let flow, addr = accept ~sw t in
+  Fun.protect
+    ~finally:(fun () ->
+      Flow.close flow;
+      Semaphore.release domains
```

Comment:
Hmm...  Assuming this `release` matches the below `acquire` call

```ocaml
    (fun () ->
      Domain_manager.run domain_mgr (fun () ->
        Semaphore.acquire domains ;
```

then this seems out of place to me unless there is a strong guarantee that `Domain_manager.run` always calls the passed function and e.g. never raises exceptions.

Reply:
Done.